lib/OpenQA/Worker/Job.pm

```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') || UPLOAD_FROM_POOL;
```

Comment:
This is not workingas intended:
```perl
my $settings = ""; 
use constant FOO => qw/ a b c d /;
my @x = (split m/,/, $settings // "") || FOO;
say "x=(@x)"
# Output: x=(a b c d)

my $settings = "a,b";
use constant FOO => qw/ a b c d /;
my @x = (split m/,/, $settings // "") || FOO;
say "x=(@x)"
# Output: x=(2)
```

Reply:
I suggest:
```perl
my @files_to_upload = split m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '';
@files_to_upload = UPLOAD_FROM_POOL unless @files_to_upload;
```

Reply:
I just tried to run following code - 
```
my @files_to_upload = split m/,/, $settings;
@files_to_upload = qw(w e r) unless @files_to_upload;
print "x=(@files_to_upload)";
# Output : x=(w e r) 
# no warnings about use of uninitialized variable ( looks like split handling this somehow ... ) 
```

so I will use your version but without `// ''` or I missing something here ?

Reply:
did you miss a `use warnings`?

```
perl -wE'
my $settings;
my @x = split m/,/, $settings;' 
Use of uninitialized value $settings in split at -e line 3.
```

I always run my oneliners with `-w`

Reply:
yeah looks like , my bad . But anyway this version is dropped already